src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -54,20 +60,157 @@ class Point(int x, int y);
                 // data class Point { }
                 Diagnostic(ErrorCode.ERR_BadRecordDeclaration, "Point").WithLocation(2, 12)
             );
-            comp = CreateCompilation(src3, parseOptions: TestOptions.RegularPreview);
+            comp = CreateCompilation(src3);
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void RecordProperties_01()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: @"
+1
+2");
+        }
+
+        [Fact]
+        public void RecordProperties_02()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public C(int a, int b)
+    {
+    }
+
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (3,13): error CS8762: There cannot be a primary constructor and a member constructor with the same parameter types.
+                // data class C(int X, int Y)
+                Diagnostic(ErrorCode.ERR_DuplicateRecordConstructor, "(int X, int Y)").WithLocation(3, 13)
+            );
+        }
+
+        [Fact]
+        public void RecordProperties_03()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public int X { get; }
+
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: @"
+0
+2");
+        }
+
+        [Fact]
+        public void RecordProperties_04()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public int X { get; } = 3;
+
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: @"
+3
+2");
+        }
+
+        [Fact]
+        public void RecordProperties_05()
+        {
+            var src = @"
+data class C(int X, int X)
+{
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (2,25): error CS0100: The parameter name 'X' is a duplicate
+                // data class C(int X, int X)
+                Diagnostic(ErrorCode.ERR_DuplicateParamName, "X").WithArguments("X").WithLocation(2, 25),
+                // (2,25): error CS0102: The type 'C' already contains a definition for 'X'
+                // data class C(int X, int X)
+                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "X").WithArguments("C", "X").WithLocation(2, 25)
```

Comment:
Seems like we should suppress one of these errors.

Reply:
Meh, I think it's OK for now. Not worth the code complexity